### ISSUE TITLE
Macro rules meta variable name raw-variable is not equal

### ISSUE THREAD

**Issue by A4-Tacks** [rust-lang/rust-analyzer#17785]
**rust-analyzer version**: rust-analyzer 1.80.0 (05147895 2024-07-21)

**rustc version**: rustc 1.80.0 (051478957 2024-07-21)

**editor or extension**: Vim 9.1 coc-rust-analyzer

**code snippet to reproduce**: 
```rust
macro_rules! x {
    ($r#break:tt) => {
        $break
    };
}
fn main() {
    x!(2);
}
```
rustc: no errors
ra: `break outside of loop (rust-analyzer E0268)`

**no error cases**:
```rust
macro_rules! x {
    ($r#break:tt) => {
        $r#break
    };
}
fn main() {
    x!(2);
}
```

```rust
macro_rules! x {
    ($break:tt) => {
        $break
    };
}
fn main() {
    x!(2);
}
```

**Comment by ChayimFriedman2** [comment]
This seems to be fixed? I cannot reproduce it.

**Comment by A4-Tacks** [comment]
> This seems to be fixed? I cannot reproduce it.

`rust-analyzer 1.82.0-nightly (7c2012d0 2024-07-26)` reproduced

![Screenshot_20240816_122510](https://github.com/user-attachments/assets/e4caecae-11fe-468d-96d8-56c7119c97cb)

**Comment by Veykril** [comment]
This does expand correctly for me as well now on the latest release

**Comment by Veykril** [comment]
We should add a test for this though I think

**Comment by lnicola** [comment]
Closed by #18561
